### ISSUE TITLE
Fatal error on `WeakBox<BaseSubscriptionListener>`

### ISSUE THREAD

**Issue by akuzminskyi** [pubnub/swift#173]
```
Duplicate elements of type 'WeakBox<BaseSubscriptionListener>' were found in a Set.
```

![SCR-20240514-eln](https://github.com/pubnub/swift/assets/6480565/2403b0ee-ac56-4865-8c4b-427ff8ad044b)
<img width="529" alt="Screenshot 2024-05-14 at 11 23 09" src="https://github.com/pubnub/swift/assets/6480565/1f898743-7333-4d79-bb1b-6fafcfc47021">

Maybe this [post](https://forums.swift.org/t/swift-set-crash-duplicate-element-found-in-set-elements-may-have-been-mutated-after-insertion/16992) can help. 

PubNub SDK Version 6.2.3(yes, this is not the latest).

**Comment by jguz-pubnub** [comment]
@akuzminskyi could you show how you add listeners to the `PubNub` instance?
And how do you store `SubscriptionListener` variables? Are they strong references?

**Comment by akuzminskyi** [comment]
> Are they strong references?
yes, and I think maybe the same listener can be trying to add twice. 

if this is the case, shouldn't we just ignore it and instead of:
```
elements.update(with: WeakBox(element))
```
>
```
elements.insert(WeakBox(element))
```

**Comment by akuzminskyi** [comment]
I think the issue is described [here](https://stackoverflow.com/questions/55674205/swift-mutable-set-duplicate-element-found). so I think this issue is with the SDK implementation.

**Comment by jguz-pubnub** [comment]
I was unable to reproduce this error on 6.2.3 when attempting to add the same listener more than once. I will investigate possible new scenarios and update you on my findings in this thread

**Comment by jguz-pubnub** [comment]
Fixed in 7.3.3
